r/tests/testthat/test-dplyr-funcs-datetime.R

```diff
@@ -711,3 +711,19 @@ test_that("am/pm mirror lubridate", {
   )
 
 })
+
+test_that("semester", {
+  test_df <- tibble(
+    month = c(1:12, NA),
+    month_char_pad = ifelse(month < 10, paste0("0", month), month),
+    dates = as.Date(paste0("2021-", month_char_pad, "-15"))
+  )
+
+  compare_dplyr_binding(
+     .input %>%
+      mutate(sem_wo_year = semester(dates),
+             sem_w_year = semester(dates, with_year = TRUE)) %>%
+      collect(),
+     test_df
+  )
```

Comment:
Would you mind also testing some bad / wrong inputs and confirming that the errors are what we expect? Something like:

```
expect_error(call_binding("semester", "not a month"), "...")
```

And other inputs like `22`, `-4`, `NA`, etc. that are clearly out of bounds / won't coerce into a month? Alternatively (or even better...) might be to add these tests to the tests for the `month` binding since that's where that coercion happens, it will be used when we make a binding for `quarters` and it appears those tests weren't added (or I can't find them!) when we added `months`

Reply:
I think the direction for this would be for `month` to support integer inputs and then all the tests to happen there. I've added a failing test for integer inputs and I will make a note / add a comment to move the test to the "it works" block once we support integer inputs for month. I have opened [ARROW-15701](https://issues.apache.org/jira/browse/ARROW-15701).

Reply:
Yup, totally agree that `month` is the thing that should do this to accept a broader array of inputs + testing there. I also commented on the Jira with some ideas about how to move forward there

Reply:
@jonkeane I decided to implement support for integer inputs in `month()`, but I do not have a working version yet.